### PR TITLE
[controller] Set `BlockOwnerDeletion` to true when create PodGroup

### LINKED ISSUE #755: Bug Report: When delete pod the related "PodGroup" won't be deleted
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

When create `Pod` with Volcano as scheduler, Volcano controller will [create](https://github.com/volcano-sh/volcano/blob/v0.3/pkg/controllers/podgroup/pg_controller.go#L134) a `PodGroup` automatically for this pod. The `OwnerReferences` field of this `PodGroup` will set to original pod. But [`BlockOwnerDeletion`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1?tab=doc#OwnerReference) field in `OwnerReference` not set, so its value will be default value (i.e. false). Reference to [official K8s document](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#controlling-how-the-garbage-collector-deletes-dependents), only object with `BlockOwnerDeletion` set to true will be deleted when in cascading deletion. So when delete original pod the related `PodGroup` won't be deleted.

**What you expected to happen**:

Delete pod will delete related `PodGroup` also.

**How to reproduce it (as minimally and precisely as possible)**:

Create a `Pod` and set `schedulerName` to `volcano`, then delete the pod see if related `PodGroup` deleted also.

**Anything else we need to know?**:

N/A

**Environment**:

- Volcano Version: v0.3
- Kubernetes version (use `kubectl version`): v1.14.6
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by xiaogaozi**
Fix #755

### CODE REVIEW

File: pkg/controllers/podgroup/pg_controller_handler.go [hunk 1]
```diff
@@ -110,11 +110,13 @@ func newPGOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
...
-		Kind:       "Pod",
-		Controller: &isController,
-		Name:       pod.Name,
-		UID:        pod.UID,
+		APIVersion:         v1.SchemeGroupVersion.Version,
+		Kind:               "Pod",
+		Controller:         &isController,
+		Name:               pod.Name,
+		UID:                pod.UID,
+		BlockOwnerDeletion: &blockOwnerDeletion,
```
Comment (hzxuzhonghu) [inline pkg/controllers/podgroup/pg_controller_handler.go:120]: nit: Instead of constructing this one by one. Can use `metav1.NewControllerRef` function.

**Comment by hzxuzhonghu** [comment]
@xiaogaozi The ut failed

```
=== RUN   TestAddPodGroup
--- FAIL: TestAddPodGroup (0.00s)
    pg_controller_test.go:157: Case AddPodGroup: pod has no ownerReferences or priorityClassName failed, expect &{{PodGroup scheduling.volcano.sh/v1beta1} {podgroup-7a09885b-b753-4924-9fba-77c0836bac20  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [{v1 Pod pod1 7a09885b-b753-4924-9fba-77c0836bac20 0xc000046250 <nil>}] []  []} {1   <nil>} { [] 0 0 0}}, got &{{ } {podgroup-7a09885b-b753-4924-9fba-77c0836bac20  test    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [{v1 Pod pod1 7a09885b-b753-4924-9fba-77c0836bac20 0xc0000462b1 0xc0000462b2}] []  []} {1   <nil>} { [] 0 0 0}}
FAIL
```

File: pkg/controllers/podgroup/pg_controller_handler.go (hunk 1 above)
Comment (xiaogaozi) [reply pkg/controllers/podgroup/pg_controller_handler.go]: Good suggestion! I've updated the PR.

**Pull request merged**